services/blockchain-connector/shared/sdk/client.js

```diff
@@ -158,14 +102,14 @@ const instantiateClient = async (isForceReInstantiate = false) => {
 		if (err.message.includes('ECONNREFUSED')) {
 			throw new Error('ECONNREFUSED: Unable to reach a network node.');
 		}
-
-		return null;
 	}
 };
 
 const getApiClient = async () => {
-	const apiClient = await waitForIt(instantiateClient, RETRY_INTERVAL);
-	return (await checkIsClientAlive()) ? apiClient : getApiClient();
+	if (cachedApiClients.length === 0) {
+		throw new Error(`No api client is alive!`);
+	}
+	return cachedApiClients[0];
```

Comment:
Instead should we create a variable counter which increments and returns different API clients?

```suggestion
	return cachedApiClients[counter++ % CONNECTION_LIMIT];
```

If we are always returning 1st client then maybe it dosen't make sense to create 5 different connections.

Reply:
It might speed up to query multiple clients simultaneously, but would also complicate the event subscription logic. I tried to prefer simplicity and reliability over perfect optimisation.